merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/types/ValueSchema.java

```diff
@@ -155,6 +166,23 @@ static ValueSchema ofVariant(final List<Variant> variants) {
     return new VariantSchema(variants);
   }
 
+  /**
+   * Augments a {@link ValueSchema} with the given metadata.
+   * If the underlying ValueSchema is already a MetaSchema, the new MetaSchema will be merged with the underlying one.
+   * In the event of a key collision, the outer MetaSchema's key shadows the inner one.
```

Comment:
Yes - in a call like `ValueSchema.withMeta("foo", SerializedValue.of("bar"), target)`, `target` is the inner one, and `foo: bar` is the "outer" one.

Let me know if you think the javadoc could be clearer about this

Reply:
I think it could be clearer if, rather than using `outer` and `inner`, the javadoc calls out explicitly that `metadata` (or something like `the new MetaSchema`) will overwrite `key` in `target`'s metadata if there's a key collision. This is a nonblocking recommendation, however.

Reply:
Reworded:

```java
  /**
   * Augments a {@link ValueSchema} with the given metadata.
   * If the underlying ValueSchema is already a MetaSchema, the new MetaSchema will be merged with the underlying one.
   * In the event of a key collision, the given `metadata` takes precedence over the target's `metadata`
   * @return A new {@link ValueSchema} wrapping the original schema.
   */
  static ValueSchema withMeta(final String key, final SerializedValue metadata, final ValueSchema target) {
```